lib/aws/AwsSigv4Signer.js

```diff
@@ -77,13 +78,23 @@ function AwsSigv4Signer(opts = {}) {
     request.region = opts.region;
     request.headers = request.headers || {};
     request.headers['host'] = request.hostname;
+    request.headers['X-Amz-Content-Sha256'] = hash(request.body || '', 'hex');
+    request.extraHeadersToIgnore = {
+      'content-length': true,
+    };
```

Comment:
Was this the issue? Content-length needed to be included?

https://github.com/opensearch-project/opensearch-js/pull/366/files

Reply:
No, looks like that PR is not including `x-amz-content-sha256` in the signing process. It signs the request first and then computes the hash in the header. Weird, I tried this but it did not work.